src/controllers/userController.ts

```diff
@@ -91,9 +91,13 @@ export class UserController {
   public async removeUser(req: Express.Request, res: Express.Response) {
     const { id } = <RemoveUserDto>matchedData(req);
     try {
-      const user = await userService.removeUser(id);
-      if (user) {
-        return res.status(StatusCodes.OK).send(user);
+      const userToRemove = await userService.removeUser(id);
+      if (userToRemove) {
+        if (req.user?.isAdmin || req.user?.id == id) {
+          return res.status(StatusCodes.OK).send(userToRemove);
+        } else {
+          return res.status(StatusCodes.FORBIDDEN).send(ReasonPhrases.FORBIDDEN);
+        }

```

Comment:
That's the second time I see these changes. If you made branch 1 with some changes, and then you made branch 2 from branch 1, when doing pr, you should do it like this:

development <= branch 1 (pr 1)
branch 1 <= branch 2. (pr 2)
Thanks to this the changes made in branch 1 will be visible only in pr 1, and people doing code review won't see them twice. I'm not sure if this is such case, but keep this in mind